lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -368,7 +370,7 @@ instance FromText SortOrder where
 data Range a = Range
     { rStart :: Maybe a
     , rEnd :: Maybe a
-    } deriving (Eq, Show)
+    } deriving (Eq, Functor, Show)
```

Comment:
Should probably replace this with a `mapRange` function, to avoid the situation where `rStart > rEnd`.